src/core/util/http_client/httpcli.cc

```diff
@@ -355,15 +394,20 @@ void HttpRequest::NextAddress(grpc_error_handle error) {
                                          &overall_error_, 1));
     return;
   }
-  const grpc_resolved_address* addr = &addresses_[next_address_++];
-  DoHandshake(addr);
+  DoHandshake(addresses_[next_address_++]);
 }
 
 void HttpRequest::OnResolved(
-    absl::StatusOr<std::vector<grpc_resolved_address>> addresses_or) {
+    absl::StatusOr<std::vector<EventEngine::ResolvedAddress>> addresses_or) {
+  ApplicationCallbackExecCtx callback_exec_ctx;
```

Comment:
I recommend moving these contexts into the EventEngine callback. For the iomgr path, these contexts must already be on the stack, so adding them here could change the ordering of callback execution.

Reply:
good point. done.